signer/jws.go

```diff
@@ -209,11 +225,27 @@ func getExtendedAttributes(attrs map[string]interface{}, critical []string) []At
 }
 
 func validateCriticalHeaders(protectedHdr *jwsProtectedHeader) error {
-	mustMarkedCrit := map[string]bool{}
+	if len(protectedHdr.Critical) == 0 {
+		return MalformedSignatureError{"missing `crit` header"}
+	}
+
+	mustMarkedCrit := map[string]bool{headerKeySigningScheme: true}
 	if protectedHdr.Expiry != nil && !protectedHdr.Expiry.IsZero() {
 		mustMarkedCrit[headerKeyExpiry] = true
 	}
 
+	if protectedHdr.SigningScheme == SigningSchemeX509SigningAuthority {
```

Comment:
For Signing scheme X509, presence of `AuthenticSigningTime` is invalid. And for X509 Signing Authority, presence of `SigningTime` is invalid. We should add those validations.

https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#signed-attributes